packages/api/app/Models/TechnologyOrder.js

```diff
@@ -14,6 +14,44 @@ class TechnologyOrder extends Model {
 	user() {
 		return this.belongsTo('App/Models/User');
 	}
+
+	/**
+	 * Runs the technology order query with the provided filters.
+	 *
+	 * @param {object} query The query object.
+	 * @param {object} request The request object.
+	 * @returns {object}
+	 */
+	static async scopeWithFilters(query, request) {
+		const filters = request.only([
+			'status',
+			'unit_value',
+			'responsible',
+			'buyer',
+			'dateStart',
+			'dateEnd',
+		]);
+		if (filters.status) {
+			const statusList = filters.status ? filters.status.split(',') : [];
+			if (statusList && statusList.length) {
+				query.whereIn('status', statusList);
+			}
+		}
+
+		if (filters.unit_value) {
+			query.where('unit_value', filters.unit_value);
+		}
+
+		if (filters.buyer) {
+			query.where('user_id', filters.buyer);
+		}
+
+		const today = new Date();
+		const initialDate = new Date('2019-01-01 00:00:00');
+		const dateStart = filters.dateStart ? filters.dateStart : initialDate;
+		const dateEnd = filters.dateEnd ? filters.dateEnd : today;
+		query.whereBetween('created_at', [dateStart, dateEnd]);
```

Comment:
Sempre aplicaremos esses filtros? Não é melhor só quando existir o `dateStart` ou `dateEnd` em `filters`?
```suggestion
		if (filters.dateStart || filters.dateEnd) {
			const dateStart = filters.dateStart || initialDate;
			const dateEnd = filters.dateEnd ||  today;
			query.whereBetween('created_at', [dateStart, dateEnd]);	
		}
```

Reply:
Boa!